atom/common/api/atom_bindings.cc

```diff
@@ -209,6 +218,60 @@ v8::Local<v8::Value> AtomBindings::GetSystemMemoryInfo(v8::Isolate* isolate,
   return dict.GetHandle();
 }
 
+// static
+v8::Local<v8::Promise> AtomBindings::GetProcessMemoryInfo(
+    v8::Isolate* isolate) {
+  scoped_refptr<util::Promise> promise = new util::Promise(isolate);
+  node::Environment* env = node::Environment::GetCurrent(isolate);
+  memory_instrumentation::MemoryInstrumentation::GetInstance()
+      ->RequestGlobalDumpForPid(base::GetCurrentProcId(),
+                                std::vector<std::string>(),
+                                base::Bind(&AtomBindings::DidReceiveMemoryDump,
+                                           base::Unretained(env), promise));
+  return promise->GetHandle();
+}
+
+// static
+void AtomBindings::DidReceiveMemoryDump(
+    node::Environment* env,
+    scoped_refptr<util::Promise> promise,
+    bool success,
+    std::unique_ptr<memory_instrumentation::GlobalMemoryDump> global_dump) {
+  // Enter Node environment.
+  mate::Locker locker(env->isolate());
+  v8::HandleScope handle_scope(env->isolate());
+  v8::MicrotasksScope script_scope(env->isolate(),
+                                   v8::MicrotasksScope::kRunMicrotasks);
+  v8::Context::Scope context_scope(env->context());
+
+  if (!success) {
+    promise->RejectWithErrorMessage("Failed to receive memory dump");
+    return;
+  }
+
+  bool resolved = false;
+  for (const memory_instrumentation::GlobalMemoryDump::ProcessDump& dump :
+       global_dump->process_dumps()) {
+    if (base::GetCurrentProcId() == dump.pid()) {
+      mate::Dictionary dict = mate::Dictionary::CreateEmpty(env->isolate());
+      const auto& osdump = dump.os_dump();
+#if defined(OS_LINUX)
+      dict.Set("residentSetBytes", osdump.resident_set_kb);
+#elif defined(OS_WIN)
+      dict.Set("workingSetSize", osdump.resident_set_kb);
```

Comment:
I decided to do that because resident set size is called working set size on windows.
Also, internally it calls the similarly named windows API which does that.

Reply:
I think the concepts are similar enough to share the name "residentSetBytes" (though it looks like it would actually be `residentSetKilobytes` from the variable name `resident_set_kb`?). Chromium calls them the same thing.

Refs:

https://docs.microsoft.com/en-us/windows/desktop/psapi/process-memory-usage-information
> The working set is the amount of memory physically mapped to the process context at a given time.

https://docs.microsoft.com/en-us/windows/desktop/procthread/process-working-set
> The working set of a program is a collection of those pages in its virtual address space that have been recently referenced. It includes both shared and private data.

https://stackoverflow.com/a/21049737
> RSS is the Resident Set Size and is used to show how much memory is allocated to that process and is in RAM. It does not include memory that is swapped out. It does include memory from shared libraries as long as the pages from those libraries are actually in memory.